includes/Story_Post_Type.php

```diff
@@ -340,4 +348,55 @@ public function update_archive_setting() {
 			flush_rewrite_rules( false );
 		}
 	}
+
+	/**
+	 * Determines whether the post type should have an archive or not.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @return bool|string Whether the post type should have an archive, or archive slug.
+	 */
+	private function get_has_archive() {
+		$archive_page_option    = get_option( Settings::SETTING_NAME_ARCHIVE );
+		$custom_archive_page_id = get_option( Settings::SETTING_NAME_ARCHIVE_PAGE_ID );
+		$has_archive            = true;
+
+		if ( 'disabled' === $archive_page_option ) {
+			$has_archive = false;
+		} elseif ( 'custom' === $archive_page_option && $custom_archive_page_id ) {
+			$uri = get_page_uri( $custom_archive_page_id );
+			if ( $uri ) {
+				$has_archive = urldecode( $uri );
+			}
+		}
+
+		return $has_archive;
+	}
+
+	/**
+	 * Modifies the current query to set up the custom archive page.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_Query $query Current query instance, passed by reference.
+	 * @return void
+	 */
+	public function pre_get_posts( WP_Query $query ) {
+		if ( ! is_string( $this->get_has_archive() ) ) {
+			return;
+		}
+
+		if ( $query->is_admin || ! $query->is_main_query() ) {
```

Comment:
Don't you need to check is `is_post_type_archive( self:POST_TYPE_SLUG );` here?

Reply:
In my testing, this didn't work. 
```
! $query->is_post_type_archive( self::POST_TYPE_SLUG )
``` 
always returned false for me.

Reply:
<img width="1884" alt="Screenshot 2021-09-21 at 17 13 49" src="https://user-images.githubusercontent.com/237508/134207753-b330a501-95ca-453f-871a-efd1456fbbb6.png">